### DISCUSSION TITLE
Icon Farbe anhand des Datenpunkt Wertes setzen

### DISCUSSION THREAD

**Discussion by djalexz** [Zefau/ioBroker.jarvis#2225] [Q&A]
Hallo zusammen,

leider finde ich keine Lösung dazu, daher die frage hier:

Ich habe einen Datenpunkt mit einem Farbwert drin z.B. BB005EFF.
Jetzt möchte ich, dass das Icon für diesen Datenpunkt in der Farbe des Datenpunkt Wertes angezeigt wird.
Mir ist klar, dass in dem Wert noch ein # fehlt, aber selbst mit dem Hinzufügen funktioniert es nicht.

Wenn ich folgendes versuche, stürzt Jarvis komplett ab:

![image](https://user-images.githubusercontent.com/11389551/217335769-e3c52c5a-1b12-4404-aba1-7a7a7f709307.png)

oder

![image](https://user-images.githubusercontent.com/11389551/217335886-2bb91338-c1c1-436c-94fe-f489d42b11d9.png)

Bitte um Hilfe

Vielen Dank

LG Alex

**Comment by mcuiobroker** [comment]
Bitte einmal das ioBroker-Objekt zeigen. Als Normales Objekt und die Objektdaten.
![image](https://user-images.githubusercontent.com/84818934/217343884-29daa649-2690-4d88-925d-7d0df7f7b31b.png)
![image](https://user-images.githubusercontent.com/84818934/217343930-3398ade3-a8d4-4b20-964c-d724cf13fa8e.png)

Alternative?
https://mcuiobroker.gitbook.io/jarvis-infos/jarvis-v3/besonderheiten-v3/geraet/actionelement/lightcolorstate

**Comment by Zefau** [comment]
Was meinst du mit jarvis stürzt ab? Bist du auf v3.1? Kannst du den Absturz mal screenshoten und auch die Browser-Konsole (F12)?

Der Hex-Code ist im übrigens falsch. Es sind nur 6 Zeichen, nicht 8.

**Comment by mcuiobroker** [comment]
CSS 8 Digit Hex Colors
![image](https://user-images.githubusercontent.com/84818934/217356645-a36399ba-389b-41c3-b520-bbab8fe96ee3.png)

**Comment by Zefau** [comment]
Jo, stimmt, hab den alpha-Kanal vergessen.

**Comment by djalexz** [comment]
Hallo zusammen,

das mit der Alternative (Lightcolorstate) ist schon mal ein guter Anfang.
![image](https://user-images.githubusercontent.com/11389551/217483876-73c8e2d4-3192-4fa8-a4a8-142c3c8cd64e.png)


Hier das Objekt um das es geht:

![image](https://user-images.githubusercontent.com/11389551/217481335-974d7cd3-90f6-4b7c-ba24-b61dffe944a3.png)


![image](https://user-images.githubusercontent.com/11389551/217481245-925e1e30-9ced-4785-af6e-b12759c2e33d.png)

Abstürzen tut es jetzt komischerweise nicht mehr.
Ich habe aber auch alle meine Testversuche und "Unfälle" wieder gelöscht.
Evtl. hatte Jarvis da etwas Schluckauf.

**Reply by djalexz** [reply]
Wenn ich aber die Alternative Lightcolorstate nutze, dann funktioniert der Style im Body Stil nicht mehr.
Also z.B. das im Body Stil: {"{0_userdata.0.X1C.print.ams.ams.0.id}=1":{"color":"red"}}

Also muss ich wohl doch es irgendwie hinbekommen, dass ich ohne Lightcolorstate arbeite.
Müsste ja eigentlich gehen oder?

Hier als Beispiel:

![image](https://user-images.githubusercontent.com/11389551/218165990-f24dee00-97c6-468d-92cf-f6dea4d67193.png)

Der Text AMS Fach 1 sollte rot werden, tut er aber nicht.
In diesem Eintrag wurde Lightcolorstate eingestellt.

Bei dem Eintrag AMS Filament Feuchtigkeit funktioniert das. (ohne Lightcolorstate ).

Ich hoffe ihr wisst was ich meine.

**Reply by mcuiobroker** [reply]
Zeig mal die Geräte-DP-Einstellungen zu dem DP mit dem Body Stil.

![image](https://user-images.githubusercontent.com/84818934/218170390-ae241b15-25f9-4171-a23e-9504085a2f6d.png)

@Zefau LightColorState nur für Hex-Werte?
#FF00FF00 -> funktioniert nicht
red -> funktioniert nicht

**Comment by djalexz** [comment]
Meinst du das?

![image](https://user-images.githubusercontent.com/11389551/218171530-842276f7-3a94-4c14-9320-8758684975a2.png)

![image](https://user-images.githubusercontent.com/11389551/218171685-f9beca8a-a570-4014-bac4-12acd815dc4a.png)

**Reply by mcuiobroker** [reply]
Nein, aber das hilft auch schon.
`{"'{0_userdata.0.X1C.print.ams.ams.0.id}'='1'":{"color":"red"}}`

color: val geht nicht

**Reply by djalexz** [reply]
Leider nicht.
Eingtragen im Body Stil.

**Reply by mcuiobroker** [reply]
Nimm mal das color beim Icon Stil raus.
Funktioniert bei mir.
![image](https://user-images.githubusercontent.com/84818934/218173029-29088ae5-ebe4-4561-ab21-fdaac0675dbc.png)
![image](https://user-images.githubusercontent.com/84818934/218173097-b2dc0fd8-d091-4d29-ab52-1146f978a03f.png)

**Reply by mcuiobroker** [reply]
Wie ist eigentlich das Gewerk vom Gerät?
Muss "LICHT" sein.

**Reply by djalexz** [reply]
Oh man ich Idiot!!!

Ich habe die ganze Zeit das falsche Gerät bearbeitet.
@mcuiobroker hat mich auf die richtige Spur gebracht.

Oh man...

Vielen lieben Dank schon mal dafür.

Für das Problem im Eingangspost gibt’s wohl keine Lösung oder nur die Alternative mit Lightcolorstate?

**Reply by mcuiobroker** [reply]
Icon und Icon Stil haben anscheinend noch kein Binding auf beiden Seiten. (key, value)
-> Feature Request?

**Comment by mcuiobroker** [comment]
`v3.1.4-beta.3`
Icon-Stil kann jetzt auch value verarbeiten
`{"color":"{value}"}`
Damit kann man den `val` in Datenpunkt-Eigenschaften bearbeiten und in Icon-Stil nutzen.
Datenpunkt-Eigenschaften
`{"value":"val=> !val ? '' : val.substr(0,val.length-2)"}`

**Reply by djalexz** [reply]
Jawohl, vielen Dank für die Info.
So habe ich mir das vorgestellt.
![image](https://user-images.githubusercontent.com/11389551/221777580-7799c069-80de-4ff1-bb66-ac69312b6333.png)

Danke euch
